From: Ricardo Neri <ricardo.neri@ti.com>
Date: Mon, 21 Feb 2011 07:01:42 +0000 (-0600)
Subject: OMAP4: HDMI: Improve readability of EDID - SVDs
X-Git-Tag: android-2.6.35-2.3-omap4.13.1~77
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=5f8b847069b65a49f70308cb0b88a3227093ead2

OMAP4: HDMI: Improve readability of EDID - SVDs

Include information about the native DTDs as well as a list of
the Short Video Descriptors in the CEA extension block and
whether they are native.

Change-Id: Id00d9d9db9f07fc54b0b1829c61f0d7d8b17f2e6
Signed-off-by: Ricardo Neri <ricardo.neri@ti.com>
---

diff --git a/arch/arm/plat-omap/include/plat/edid.h b/arch/arm/plat-omap/include/plat/edid.h
index d6f2894..4e76563 100644
--- a/arch/arm/plat-omap/include/plat/edid.h
+++ b/arch/arm/plat-omap/include/plat/edid.h
@@ -78,6 +78,7 @@
 #define HDMI_EDID_DTD_TAG_ASCII_STRING		0xFE
 
 #define HDMI_IMG_FORMAT_MAX_LENGTH		20
+#define HDMI_VIDEO_NATIVE_DTDS_MASK		0x0f
 #define HDMI_AUDIO_FORMAT_MAX_LENGTH		10
 #define HDMI_AUDIO_BASIC_MASK			0x40
 
diff --git a/drivers/video/omap2/dss/hdmi.c b/drivers/video/omap2/dss/hdmi.c
index 480b785..c156853 100644
--- a/drivers/video/omap2/dss/hdmi.c
+++ b/drivers/video/omap2/dss/hdmi.c
@@ -1952,7 +1952,7 @@ static struct hdmi_cm hdmi_get_code(struct omap_video_timings *timing)
 static void hdmi_get_edid(struct omap_dss_device *dssdev)
 {
 	u8 i = 0, mark = 0, *e;
-	int offset, addr;
+	int offset, addr, length;
 	struct HDMI_EDID *edid_st = (struct HDMI_EDID *)edid;
 	struct image_format *img_format;
 	struct audio_format *aud_format;
@@ -2056,10 +2056,21 @@ static void hdmi_get_edid(struct omap_dss_device *dssdev)
 				dss_debug = mark;
 			}
 		}
+		hdmi_get_video_svds(edid, &offset, &length);
+		printk(KERN_INFO "No of SVDs: %d", length);
+		for (i = 0; i < length; i++) {
+			printk(KERN_INFO "SVD[%d]: CEA code[%d], native[%s]",
+				i, edid[offset+i] & HDMI_EDID_EX_VIDEO_MASK,
+				edid[offset+i] & HDMI_EDID_EX_VIDEO_NATIVE ?
+				"YES" : "NO");
+		}
+		printk(KERN_INFO "No. of native DTD: %d",
+				(edid[EDID_DESCRIPTOR_BLOCK1_ADDRESS + 3]
+			& HDMI_VIDEO_NATIVE_DTDS_MASK));
+
 		printk(KERN_INFO "Supports basic audio: %s",
 				(edid[EDID_DESCRIPTOR_BLOCK1_ADDRESS + 3]
 			& HDMI_AUDIO_BASIC_MASK) ? "YES" : "NO");
-
 	}
 
 	printk(KERN_INFO "Has IEEE HDMI ID: %s",